src/sentry/search/events/datasets/spans_metrics.py

```diff
@@ -566,6 +566,30 @@ def function_converter(self) -> Mapping[str, fields.MetricsFunction]:
                     snql_distribution=self._resolve_count_op,
                     default_result_type="integer",
                 ),
+                fields.MetricsFunction(
+                    "trace_status_rate",
+                    required_args=[
+                        SnQLStringArg("status"),
+                    ],
+                    snql_distribution=lambda args, alias: function_aliases.resolve_division(
+                        self._resolve_trace_status_count(args),
+                        Function(
+                            "countIf",
+                            [
+                                Column("value"),
+                                Function(
+                                    "equals",
+                                    [
+                                        Column("metric_id"),
+                                        self.resolve_metric("span.self_time"),
+                                    ],
+                                ),
+                            ],
+                        ),
```

Comment:
To make this more robust, should we only count if there is a trace.status?

Reply:
Hmm, good point. I thought about this for a bit and I _think_ there might scenarios where we want to include spans that don't have a `trace.status`. For cases where we only want to count spans that have a `trace.status`, we should be able to just add something like `has:trace.status` to the discover query to achieve that affect, so I think I'm going to leave this as is for the flexibility.